### PR TITLE
Add MineCard

### PR DESCRIPTION
**Pull request by jakemagner**

### CODE REVIEW

File: base_set/cards/mine.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from core.card import CardType
+from core.card import KingdomCard
+from core.card import TreasureCard
+from core.decision import Decision
+from core.locations import Location
+from core.locations import LocationName
+
+
+class MineTrashDecision(Decision):
```
Comment (michelleon) [inline base_set/cards/mine.py]: should both the trash and gain happen at the same time? I feel like agents will calculate them at the same time. do they have to hold their state in memory or something and hang until the next decision

File: base_set/cards/mine.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.options = treasure_cards
+        self.min = 0
+        self.max = 1
+
+
+class MineGainDecision(Decision):
+    def __init__(self, game_state, max_cost):
+        player = game_state.get_current_player_name()
+        cards = set(game_state.get_location(Location(None, LocationName.SUPPLY)))
+        treaure_cards_meeting_cost_constraint = [
```
Comment (michelleon) [inline base_set/cards/mine.py]: typo: `treasure`

File: tests/mine_card_test.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            if GoldCard in decision.options:
+                return [GoldCard]
+            elif SilverCard in decision.options:
+                return [SilverCard]
+            else:
+                return decision.choose_random()
+        return decision.choose_random()
+
+
+class MineCardTest(unittest.TestCase):
```
Comment (michelleon) [inline tests/mine_card_test.py]: tests! :D

File: base_set/cards/mine.py (hunk 1 above)
Comment (jakemagner) [reply base_set/cards/mine.py]: We could compute every combination of (card_to_trash, card_to_gain) and pass those options. Still, other sets may cause something to happen in between the trash and the gain. Like isn't there some card that says when you trash something you may discard it instead, so we'd have to then insert that decision in between.

**Comment by michelleon** [comment]
LGTM

**Pull request merged**